packages/client/src/views/OIDPVerificationCallback/utils.ts

```diff
@@ -196,19 +199,38 @@ function splitName(name: string | undefined | null = '') {
   }
 }
 
-export function saveDraftAndRedirectToNidIntegration(
-  declaration: IDeclaration | IPrintableDeclaration,
-  writeDeclration: (
-    declaration: IDeclaration | IPrintableDeclaration,
-    callback?: (() => void) | undefined
-  ) => void,
+export const useNidAuthentication = () => {
+  const dispatch = useDispatch()
+  const match = useRouteMatch()
+  const matchParams = match.params as {
+    declarationId: string
+    groupId: string
+    pageId: string
+  }
+  const offlineCountryConfig = useSelector(getOfflineData)
+  const declaration = useDeclaration(matchParams.declarationId)
+
+  const onClick = () =>
+    dispatch(
+      writeDeclaration(declaration, () => {
+        redirectToNidIntegration(
+          offlineCountryConfig,
+          matchParams.declarationId,
+          matchParams.pageId,
+          match.url
+        )
+      })
+    )
+
+  return { onClick }
+}
+
+export function redirectToNidIntegration(
   offlineCountryConfig: IOfflineData,
   declarationId: string,
   currentSection: string,
   currentPathname: string
 ) {
-  writeDeclration(declaration)
-
```

Comment:
I don't know with what kind of holy spirit this has worked earlier, because it's not connected or dispatched with Redux at all.